lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -2122,9 +2147,18 @@ ParseResult FIRStmtParser::parseInstance() {
     resultTypes.push_back(FlipType::get(port.type));
     resultNames.push_back(port.name);
   }
+  auto name = filterUselessName(id);
+  auto parentModule =
+      dyn_cast_or_null<FModuleOp>(builder.getBlock()->getParentOp());
+  ArrayAttr annotations = builder.getArrayAttr({});
+  if (parentModule) {
+    auto target = "~" + circuit.name().str() + "|" +
+                  parentModule.getName().str() + ">" + name.getValue().str();
+    getAnnotations(target, annotations);
```

Comment:
Please try to reduce the cost (string concats etc) in the case where there are no annotations.  I'm not sure what is possible here though.

Reply:
Switching to `Twine` in the rebase should make the concats way cheaper. I'm also adding an early exit to the `getAnnotations` method if no annotations were parsed in. (This won't save much overall, but avoids unnecessary work.)

Further improvements I'll have to think about...